extutils/locales.py

```diff
@@ -21,122 +26,257 @@ def sec_diff_to_utc_offset(s_diff: float):
 
-        If ``tzname`` is ``None`` or timezone not found and ``silent_fail`` is ``True``, return ``None``.
+        Reference: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+
+        :param tzidentifier: timezone identifier defined in IANA timezone database
+        :param silent_fail: if to silently fail if not found
+        :return: a `PytzInfo` if timezone found
         """
-        if tzname is None:
+        if tzidentifier is None:
             return None
 
         try:
-            return PytzInfo(pytz.timezone(tzname))
-        except pytz.exceptions.UnknownTimeZoneError as e:
+            return PytzInfo(pytz.timezone(tzidentifier))
+        except pytz.exceptions.UnknownTimeZoneError as ex:
             if silent_fail:
                 return None
 
-            raise e
+            raise ex
 
 
 @dataclass
 class LanguageInfo:
+    """Class representing a language info."""
+
     name: str
     code: str
 
 
 class PytzInfo(tzinfo):
+    """A customized :class:`tzinfo` based on the :class:`tzinfo` from ``pytz``."""
+
     def __init__(self, tz):
         self._base = tz
 
     def utcoffset(self, dt):
         if self._base is pytz.UTC:
             # UTC does not accept `is_dst`
             return self._base.utcoffset(dt.replace(tzinfo=None))
-        else:
-            return self._base.utcoffset(dt.replace(tzinfo=None), is_dst=is_now_dst(self._base))
+
+        return self._base.utcoffset(dt.replace(tzinfo=None), is_dst=is_now_dst(self._base))
 
     def dst(self, dt):
         if self._base is pytz.UTC:
             # UTC does not accept `is_dst`
             return self._base.dst(dt.replace(tzinfo=None))
-        else:
-            return self._base.dst(dt.replace(tzinfo=None), is_dst=is_now_dst(self._base))
+
+        return self._base.dst(dt.replace(tzinfo=None), is_dst=is_now_dst(self._base))
 
     def tzname(self, dt):
         return sec_diff_to_utc_offset(self.utcoffset(dt).total_seconds())
 
-    def localize(self, tz, *, is_dst: bool = False):
-        return self._base.localize(tz, is_dst)
+    def localize(self, dt: datetime, *, is_dst: bool = False) -> datetime:
+        """
+        Localize the datetime ``dt``.
+
+        ``dt`` must be tz-naive.
+
+        Uses ``is_dst`` to determine whether DST is in effect if unable to determine.
+
+        :param dt: datetime to be localized
+        :param is_dst: default setting of DST if unable to determine
+        :return: a localized `datetime`
+        :raises ValueError: if `dt` is not tz-naive
+        """
+        return self._base.localize(dt, is_dst)
 
     @property
     def tzidentifier(self):
+        """
+        Returns the ``pytz`` timezone identifier of this :class:`tzinfo`.
+
+        :return: pytz timezone identifier
+        """
         return self._base.zone
 
     def __eq__(self, other):
-        return type(self) == type(other) and self.tzidentifier == other.tzidentifier
+        return type(self) == type(
```

Comment:
Codacy found an issue: [Using type() instead of isinstance() for a typecheck.](https://app.codacy.com/manual/RaenonX/Jelly-Bot/pullRequest?prid=6015490)